ServiceApp/src/org/droidplanner/services/android/core/gcs/follow/Follow.java

```diff
@@ -103,11 +123,38 @@ public boolean isEnabled() {
         return state == FollowStates.FOLLOW_RUNNING || state == FollowStates.FOLLOW_START;
     }
 
+    public FollowLocationSource getLocationSource() {
+        return mLocationSource;
+    }
+
+    public void setLocationSource(FollowLocationSource source) {
+        if(mLocationSource != source) {
+            switch(source) {
+                case CLIENT_SPECIFIED: {
+                    Timber.d("Switch to client-specified locations");
+                    locationFinder.removeLocationListener(TAG);
+                    locationFinder.disableLocationUpdates();
+                    break;
+                }
+
+                default: {
```

Comment:
@kellyschrock the `default` label may be confusing, since from the context here it's referring to the `FollowLocationSource.INTERNAL` enum, but the class default value for the `mLocationSource` field is `FollowLocationSource.NONE`.
Let's remove the `default` label from the switch block, and add the `case INTERNAL` label for this block instead.